src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -150,41 +151,74 @@ private void addListingToBusinesses(PlacesSearchResult place, String url) {
           placeLocation, rating, photos, types, url));
   }
 
-  public void checkIfBusinessesAreBig() {
-    GeoApiContext context = 
-      new GeoApiContext.Builder().apiKey(KEY).build();
+  public void determineIfTheBusinessesAreBig() {
```

Comment:
I think you have two methods that imply they're doing the same work, but they're acting independently: removeBigBusinessesFromResults(), and determineIfBusinessesAreBig(). It seems, to me that determineIfBusinessesAreBig() should be a call from removeBigBusinessesFromResults(). i.e
```
removeBigBusinessesFromResults() {
  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
  Iterator<Listing> businesses =  allBusinesses.iterator();
  String businessName;
  
  while (businesses.hasNext()) {
   Listing currentBusiness = businesses.next();
   try {
        businessName = 
            (String) datastore.get(KeyFactory.createKey(
                                    BIG_BUSINESSES_DATABASE, 
                                    currentBusiness.getName()))
                              .getProperty("Business");

    if (businessName.equals(currentBusiness.getName()) || determineIfTheBusinessesAreBig(currentBusiness) {
        businesses.remove(currentBusiness)
    }
....
}
```

With that determineIfTheBusinessesAreBig() should be refactored to only consider one business at a time. This will also allow you to remove the while loop that's in determineIfTheBusinessAreBig() because the notion of currentBusiness will then be passed in as a parameter.

Reply:
Following that, determineIfBusinessesAreBig() can become a simple wrapper method for your actual checks: checkIfSmallBusinessInDatabase(), checkNumberOfSimilarBusinesses(), checkBusinessThroughLinkedin():
```
private boolean determineIfBussinessIsBig(Listing currentBusiness) {
    return (checkIfSmallBusinessInDatabase(currentBusiness) &&
        checkNumberOfSimilarBusinesses(currentBusiness) && 
        checkBusinessThroughLinkedin(currentBusiness))
}
```

Reply:
I can not write the if statement like you did because, since I am using a try catch block. If the big business is not in the database it follows the code inside the catch statement. So I decided to make a if statement that uses determineIfTheBusinessesAreBig() to delete the business in the same way. 
`catch(EntityNotFoundException e){
    if(determineIfTheBusinessesAreBig(currentBusiness)) {
      businesses.remove();
     }
  }`

However, the problem with what you wrote within determineIfTheBusinessesAreBig() is that I do not want to do the other checks if the business is found in the SmallBusiness database. In order to do this, I used a if else statement as follows: 
`  public boolean determineIfTheBusinessesAreBig(Listing currentBusiness) {
     boolean isBigBusiness = false;
     if(checkIfSmallBusinessInDatabase(currentBusiness)) {
       isBigBusiness = false;
     }else {
       isBigBusiness = checkNumberOfSimilarBusinesses(currentBusiness);
     }
     return isBigBusiness;
  }`
Also, since the checks are dependent of each other, I only should call one as they call each other if one check is not sufficient enough to determine if a business is big.

Reply:
I discussed this more with Shaquille and highlighted what I changed in the program in the description of the pull request - https://github.com/googleinterns/SmallCity/pull/77#issue-457938649